src/classlibnative/bcltype/system.cpp

```diff
@@ -99,12 +99,142 @@ FCIMPL0(INT64, SystemNative::__GetSystemTimeAsFileTime)
 FCIMPLEND;
 
 
+#ifndef FEATURE_PAL
+typedef int (WINAPI *NtQuerySystemInformationProc) (int SystemInformationClass, PVOID SystemInformation, ULONG SystemInformationLength, PULONG ReturnLength OPTIONAL);
+typedef struct _SYSTEM_LEAP_SECOND_INFORMATION {
+    BOOLEAN Enabled;
+    ULONG Flags;
+} SYSTEM_LEAP_SECOND_INFORMATION, *PSYSTEM_LEAP_SECOND_INFORMATION;
+
+#define SystemLeapSecondInformation 206
+
+
+BOOL QCALLTYPE SystemNative::IsLeapSecondsSupportedSystem()
+{
+    QCALL_CONTRACT;
+
+    BOOL ret = FALSE;
+
+    BEGIN_QCALL;
+
+    SYSTEM_LEAP_SECOND_INFORMATION slsi;
+    HINSTANCE hNtDll = CLRLoadLibrary(W("ntdll.dll"));
+
+    if (hNtDll != NULL)
+    {
+        NtQuerySystemInformationProc pfnNtQuerySystemInformation = (NtQuerySystemInformationProc) GetProcAddress(hNtDll,"NtQuerySystemInformation");
+        if (pfnNtQuerySystemInformation != NULL)
+        {
+            if (pfnNtQuerySystemInformation(SystemLeapSecondInformation, &slsi, sizeof(SYSTEM_LEAP_SECOND_INFORMATION), NULL) == 0)
+            {
+                ret = slsi.Enabled;
+            }
+        }
+    }
+
+    END_QCALL;
+
+    return ret;
+}
+
+FCIMPL1(VOID, SystemNative::GetSystemTimeWithLeapSecondsHandling, FullSystemTime *time)
+{
+    FCALL_CONTRACT;
+    INT64 timestamp;
+
+    g_pfnGetSystemTimeAsFileTime((FILETIME*)&timestamp);
+
+    if (::FileTimeToSystemTime((FILETIME*)&timestamp, (LPSYSTEMTIME) time))
+    {
+#if BIGENDIAN
+        timestamp = (INT64)(((UINT64)timestamp >> 32) | ((UINT64)timestamp << 32));
+#endif
+        // to keep the time precision
+        time->hundredNanoSecond = timestamp % 10000; // 10000 is the number of 100-nano seconds per Millisecond
+    }
+    else
+    {
+        ::GetSystemTime((LPSYSTEMTIME) time);
+        time->hundredNanoSecond = 0;
+    }
+
+    if (time->wSecond > 59)
+    {
+        // we have a leap second, force it to last second in the minute as DateTime doesn't account for leap seconds in its calculation.
+        // we use the maxvalue from the milliseconds and the 100-nano seconds to avoid reporting two out of order 59 seconds
+        time->wSecond = 59;
+        time->wMillisecond = 999;
+        time->hundredNanoSecond = 9999;
+    }
+}
+FCIMPLEND;
+
+FCIMPL2(FC_BOOL_RET, SystemNative::SystemFileTimeToSystemTime, INT64 fileTime, FullSystemTime *time)
+{
+    FCALL_CONTRACT;
+    INT64 timestamp = 0;
+
+    if (::FileTimeToSystemTime((FILETIME*)&fileTime, (LPSYSTEMTIME) time))
+    {
+
+#if BIGENDIAN
```

Comment:
You can delete `BIGENDIAN` ifdefs in Windows-specific code. Windows is never going to run on big endian processors.